hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataLogRecordReader.java

```diff
@@ -253,7 +253,7 @@ public HoodieMetadataLogRecordReader build() {
     }
 
     private boolean shouldUseMetadataMergedLogRecordScanner() {
-      return PARTITION_NAME_SECONDARY_INDEX.equals(partitionName);
+      return partitionName.startsWith(PARTITION_NAME_SECONDARY_INDEX_PREFIX);
```

Comment:
note to reviewer: this is amin fix in the latest commit. The issue was not caught in the testing previously because I was only asserting count of records. I have improved the test. Now, we assert the secondary index records as well as check that file pruning happens. Please see `TestSecondaryIndexPruning`